artiq/firmware/runtime/moninj.rs

```diff
@@ -18,23 +18,23 @@ mod local_moninj {
             csr::rtio_moninj::mon_chan_sel_write(channel as _);
             csr::rtio_moninj::mon_probe_sel_write(probe);
             csr::rtio_moninj::mon_value_update_write(1);
-            csr::rtio_moninj::mon_value_read() as u32
+            csr::rtio_moninj::mon_value_read()
         }
     }
 
-    pub fn inject(channel: u16, overrd: u8, value: u8) {
+    pub fn inject(channel: u16, overrd: u8, value: u32) {
         unsafe {
             csr::rtio_moninj::inj_chan_sel_write(channel as _);
             csr::rtio_moninj::inj_override_sel_write(overrd);
-            csr::rtio_moninj::inj_value_write(value);
+            csr::rtio_moninj::inj_value_write(value.to_be());
         }
     }
 
-    pub fn read_injection_status(channel: u16, overrd: u8) -> u8 {
+    pub fn read_injection_status(channel: u16, overrd: u8) -> u32 {
         unsafe {
             csr::rtio_moninj::inj_chan_sel_write(channel as _);
             csr::rtio_moninj::inj_override_sel_write(overrd);
-            csr::rtio_moninj::inj_value_read()
+            csr::rtio_moninj::inj_value_read().to_be()
```

Comment:
We only have little-endian CPUs in ARTIQ now. What is this for? If this is for the network protocol it doesn't belong here.